channeldb/invoice_test.go

```diff
@@ -99,7 +99,10 @@ func TestInvoiceWorkflow(t *testing.T) {
 	// now have the settled bit toggle to true and a non-default
 	// SettledDate
 	payAmt := fakeInvoice.Terms.Value * 2
-	if _, err := db.AcceptOrSettleInvoice(paymentHash, payAmt); err != nil {
+	_, err = db.AcceptOrSettleInvoice(
```

Comment:
I think this commit could've easily been split up into multiple commits. There's a lot going on in this commit across multiple files and packages.

Reply:
How would you do this easily? This commit is a set of changes that are all interconnected. Leaving some out will not result in a working `lnd`. I tried to split off all independent changes already. Also, I want to fix tests in the same commit that breaks them, so those can't be moved out.

Reply:
Yeah there's a tradeoff w.r.t having them always compile vs having them make smaller changes at a time with an eye for ease of review. IMO things are more likely to slip by with larger commits that make modifications over several packages/files. Isolated commits also allow the committer to tell a narrative in a sense, with the progression of each commit. 

As it's already received some attention as is, I think it's too late at this point. However, as an example, the `AcceptOrSettleInvoice` change could've been added in a distinct commit that just left all the logic in place, but used a `nil` value for the function closure in the  link. A follow up commit could then remove the old verification logic and add the height based verification to that new closure.

Reply:
Ok, that is a good example indeed. Could have been a separate commit. Not sure about committing non-building code though.